go/pkg/client/tree_test.go

```diff
@@ -1071,8 +1072,8 @@ func TestComputeMerkleTree(t *testing.T) {
 			wantCacheCalls: map[string]int{
 				"foobarDir/foo": 2, // 1 via the dir and 1 via the symlink
 				"foobarDir/bar": 2,
-				"foobarDir":     3, // 1 as input, 2 as a real ancestor
-				"foobarSymDir":  3, // 1 as input, 2 as a symlink ancestor
+				"foobarDir":     3, // 1 as target of the symlink, 2 as input ancestor
+				"foobarSymDir":  4, // 1 as input, 1 as added symlink ancestor, 2 as input ancestor
```

Comment:
nit: some tests use foobarSymDir->foobarDir and some use foobarDir->foobarDirOrig. Can we make it consistent?

Reply:
It's a compromise to avoid creating more global digests in this test. Basically this allows `foobarDir` the symlink to materialize as a directory whose digest is already known.